src/data/bills/utils/merge.ts

```diff
@@ -0,0 +1,59 @@
+import { DTSI_Bill } from '@/data/dtsi/generated'
+import { SWCBill } from '@/utils/shared/zod/getSWCBills'
+
+export type DTSI_BillDetails = Pick<
+  DTSI_Bill,
+  | 'computedStanceScore'
+  | 'congressDotGovUrl'
+  | 'dateIntroduced'
+  | 'shortTitle'
+  | 'slug'
+  | 'status'
+  | 'summary'
+  | 'title'
+> & { analysis: { richTextCommentary: string }[] }
+
+export function mergeBillFromBuilderIOAndDTSI(
+  billFromBuilderIO: SWCBill,
+  billFromDTSI: DTSI_BillDetails | null | undefined,
+): SWCBill {
+  if (!billFromDTSI) {
+    return billFromBuilderIO
+  }
+
+  return {
+    ...billFromBuilderIO,
+    analysis:
+      billFromBuilderIO.analysis ||
+      billFromDTSI.analysis?.reduce(
+        (analysis, { richTextCommentary }) => analysis + richTextCommentary,
+        '',
+      ) ||
+      '',
+    computedStanceScore: billFromDTSI.computedStanceScore,
+    dateIntroduced: billFromBuilderIO.dateIntroduced || billFromDTSI.dateIntroduced,
+    dtsiSlug: billFromBuilderIO.dtsiSlug || billFromDTSI.slug,
+    officialBillUrl: billFromBuilderIO.officialBillUrl || billFromDTSI.congressDotGovUrl,
+    summary: billFromBuilderIO.summary || billFromDTSI.summary,
+    title: billFromBuilderIO.title || billFromDTSI.shortTitle || billFromDTSI.title,
+  }
+}
+
+export function mergeBillsFromBuilderIOAndDTSI(
+  billsFromBuilderIO: SWCBill[],
+  billsFromDTSI: DTSI_BillDetails[] | null | undefined,
+): SWCBill[] {
+  if (!billsFromDTSI || billsFromDTSI.length === 0) {
+    return billsFromBuilderIO
+  }
+
+  const mergedBills: SWCBill[] = []
+
+  for (const billFromBuilderIO of billsFromBuilderIO) {
+    const billFromDTSI = billsFromDTSI.find(bill => bill.slug === billFromBuilderIO.dtsiSlug)
+
+    mergedBills.push(mergeBillFromBuilderIOAndDTSI(billFromBuilderIO, billFromDTSI))
+  }
+
+  return mergedBills
```

Comment:
np: this could be written as a `.map()` call
```suggestion
  return billsFromBuilderIO.map(billFromBuilderIO => {
    const billFromDTSI = billsFromDTSI.find(bill => bill.slug === billFromBuilderIO.dtsiSlug)
    
    return mergeBillFromBuilderIOAndDTSI(billFromBuilderIO, billFromDTSI)
  })
```